python/dfu.py

```diff
@@ -21,6 +21,7 @@ def __init__(self, dfu_serial):
         except Exception:
           continue
         if this_dfu_serial == dfu_serial or dfu_serial is None:
+          self._hw_gen = self.get_hw_generation(device)
```

Comment:
this doesn't work, does it? the device is already in dfu mode?

Reply:
This checks generation when in DFU mode by bcdDevice value